pkg/driver/aws-ebs/aws_ebs_tags_controller.go

```diff
@@ -0,0 +1,441 @@
+			return false, nil
+		}
+		if infra != nil {
+			return true, nil
+		}
+		return false, nil
+	})
+	return infra, err
+}
+
+func (c *EBSVolumeTagsController) getEBSCloudCredSecret(ctx context.Context) (*v1.Secret, error) {
+	backoff := wait.Backoff{
+		Duration: operationDelay,
+		Factor:   operationBackoffFactor,
+		Steps:    operationRetryCount,
+	}
+	var awsCreds *v1.Secret
+	err := wait.ExponentialBackoffWithContext(ctx, backoff, func(ctx context.Context) (bool, error) {
+		var apiError error
+		awsCreds, apiError = c.commonClient.KubeClient.CoreV1().Secrets(awsEBSSecretNamespace).Get(ctx, awsEBSSecretName, metav1.GetOptions{})
+		if apiError != nil {
+			klog.Errorf("error getting secret object: %v", apiError)
+			return false, nil
+		}
+		if awsCreds != nil {
+			return true, nil
+		}
+		return false, nil
+	})
+	return awsCreds, err
+
+}
+
+// processInfrastructure processes the Infrastructure resource and updates EBS tags
+func (c *EBSVolumeTagsController) processInfrastructure(ctx context.Context, infra *configv1.Infrastructure, ec2Client *ec2.EC2) error {
+	if infra.Status.PlatformStatus != nil && infra.Status.PlatformStatus.AWS != nil &&
+		infra.Status.PlatformStatus.AWS.ResourceTags != nil {
+		awsInfra := infra.Status.PlatformStatus.AWS
+		err := c.fetchPVsAndUpdateTags(ctx, awsInfra.ResourceTags, ec2Client)
+		if err != nil {
+			klog.Errorf("Error processing PVs for infrastructure update: %v", err)
+			return err
+		}
+	}
+	return nil
+}
+
+// fetchPVsAndUpdateTags retrieves all PVs and updates the AWS EBS tags in batches of 100
+func (c *EBSVolumeTagsController) fetchPVsAndUpdateTags(ctx context.Context, resourceTags []configv1.AWSResourceTag, ec2Client *ec2.EC2) error {
+	pvs, err := c.listPersistentVolumesWithRetry(ctx)
+	if err != nil {
+		return fmt.Errorf("error fetching PVs: %v", err)
+	}
+	// Compute the hash for the new set of tags
+	newTagsHash := computeTagsHash(resourceTags)
+	pvsToBeUpdated := filterUpdatableVolumes(pvs, newTagsHash)
+
+	// If there are no volumes to update, return early
+	if len(pvsToBeUpdated) == 0 {
+		klog.Infof("No volume tags to update as hashes are unchanged")
+		return nil
+	}
+
+	// Process the volumes in batches
+	for i := 0; i < len(pvsToBeUpdated); i += batchSize {
+		end := i + batchSize
+		if end > len(pvsToBeUpdated) {
+			end = len(pvsToBeUpdated)
+		}
+		batch := pvsToBeUpdated[i:end]
+
+		// Update tags on AWS EBS volumes
+		err = c.updateEBSTags(batch, ec2Client, resourceTags)
+		if err != nil {
+			c.handleTagUpdateFailure(batch, err)
+			continue
+		}
+
+		// Update PV annotations after successfully updating the tags in AWS
+		for _, volume := range batch {
+			// Set the new tag hash annotation in the PV object
+			setPVTagHash(volume, newTagsHash)
+
+			// Update the PV with the new annotations
+			err = c.updateVolumeWithRetry(ctx, volume)
+			if err != nil {
+				klog.Errorf("Error updating PV annotations for volume %s: %v", volume.Name, err)
```

Comment:
If we are already going to handle these PVs via `failedWorkerQueue`, why does the code in `updateVOlumeWithRetry` needs to have its own local exponential backoff?